### ISSUE TITLE
Share successful experience in configuring TimeSuite environment  (March is the  month learning from Lei Feng)

### ISSUE THREAD

**Issue by LiangMeng89** [OpenGVLab/TimeSuite#3]
@LanXingXuan @shepnerd @orashi @czczup @JustinYuu @ZhenhangHuang 
Thank you for your open source sharing. I encountered a lot of dependency conflicts in the process of configuring your shared TimeSuite project environment, which took a lot of time. I share with you the process of finally configuring the environment, so that later comers can save time.

*******************************************************************************************************************

Create a new environment and run the command to install the necessary dependencies:
conda create -n TimeSuite python=3.9 -y
conda activate TimeSuite
pip install -r requirements.txt / pip install -r requirements.txt -i https://pypi.tuna.tsinghua.edu.cn/simple

********************************************************************************************************************
requirements.txt:
"""
250317.
Originated from TimeSuite team, 
lm@ZJU modified,Tel/wechat:19357260600,liangmeng89@zju.edu.cn. 
Refered to the environment configuration of video_chat2 -->
https://github.com/OpenGVLab/Ask-Anything/tree/main/video_chat2 (Prepare the envirment & requirements.txt).
"""
--extra-index-url https://download.pytorch.org/whl/cu117
torch==1.13.1+cu117
torchvision==0.14.1+cu117
torchaudio==0.13.1

#------------------------------------------------------- Conflict modification -------------------------------------------------#
awscli==1.32.87
#awscli>=1.34.91  # 需要与 botocore>=1.34.91 兼容

#boto3==1.34.91
boto3==1.34.87
#botocore==1.34.91
botocore==1.34.87

#decorator==5.1.1
decorator==4.4.2  # 兼容 moviepy 的约束

fastapi==0.110.0

flash-attn==2.5.6    # 参考video_chat2环境配置,官网下载2.3.5 CU117 CP39版本安装，但被2.5.6替换

gast==0.5.4
#gast>=0.2.1,<=0.4.0  # 兼容 TensorFlow 2.13.1

keras==3.1.1
#keras>=2.13.1,<2.14  # 匹配 TensorFlow 2.13.1 的约束

libavcodec-extra     # ffmpeg的依赖项，推理时不安装会报错

#mkl-fft==1.3.1
mkl-fft==1.3.11  # 实现高性能N维傅立叶变换的Python接口,依赖numpy

#numpy==1.23.5
#numpy==1.24.4   # 参考video_chat2环境配置
numpy==1.26.4    # mkl-fft==1.3.11依赖

pydantic_core==2.16.3

#requests-oauthlib==1.4.0
requests-oauthlib==2.0.0

tensorboard==2.16.2

tensorflow==2.16.1
#tensorflow==2.13.1

typing_extensions==4.12.2

#urllib3==2.2.3
urllib3==1.26.20

#webcolors==1.13
webcolors==24.11.1  # 满足 jsonschema[format-nongpl] 的依赖:webcolors>=24.6.0

#------------------------------------------------------ keep immovability -------------------------------------------------------#
absl-py==2.1.0
accelerate==0.28.0
addict==2.4.0
aiofiles==23.2.1
aiohttp==3.9.3
aiosignal==1.3.1
aliyun-python-sdk-core==2.15.0
aliyun-python-sdk-kms==2.16.2
altair==5.2.0
annotated-types==0.6.0
antlr4-python3-runtime==4.9.3
anyio==4.6.0
anykeystore==0.2
apex==0.9.10.dev0
appdirs==1.4.4
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asttokens==2.4.1
astunparse==1.6.3
async-lru==2.0.4
async-timeout==4.0.3
attrs==24.2.0
audioread==3.0.1
av==10.0.0

Babel==2.14.0
beautifulsoup4==4.12.3
bitsandbytes==0.42.0
bleach==6.1.0
blis==0.7.11

braceexpand==0.1.7
catalogue==2.0.10
certifi==2024.8.30
cffi==1.17.1
charset-normalizer==3.3.2
click==8.1.7
cloudpathlib==0.16.0
colorama==0.4.4
coloredlogs==15.0.1
comm==0.2.2
conda-pack==0.7.1
confection==0.1.4
contourpy==1.2.0
crcmod==1.7
cryptacular==1.6.2
cryptography==42.0.5
cycler==0.12.1
cymem==2.0.8
datasets==2.19.0
debugpy==1.8.1

decord==0.6.0
deepspeed==0.14.0
defusedxml==0.7.1
dill==0.3.8
docker-pycreds==0.4.0
docutils==0.16
einops==0.6.1
environs==11.0.0
exceptiongroup==1.2.2
executing==2.1.0
fairscale==0.4.13

fastjsonschema==2.19.1
ffmpeg==1.4
ffmpy==0.3.2
filelock==3.13.1

flatbuffers==24.3.7
fonttools==4.50.0
fqdn==1.5.1
frozenlist==1.4.1
fsspec==2024.3.1
ftfy==6.2.0
fvcore==0.1.5.post20221221

gitdb==4.0.11
GitPython==3.1.42
google-pasta==0.2.0
gradio==3.35.0
gradio_client==0.12.0
greenlet==3.0.3
grpcio==1.62.1
h11==0.14.0
h5py==3.10.0
hjson==3.1.0
httpcore==1.0.5
httpx==0.27.2
huggingface-hub==0.21.4
humanfriendly==10.0
humanize==4.9.0
hupper==1.12.1
idna==3.10
imageio==2.27.0
imageio-ffmpeg==0.4.9
importlib_metadata==8.5.0
importlib_resources==6.3.1
iopath==0.1.10
ipykernel==6.29.3
ipython==8.18.1
ipywidgets==8.1.2
isoduration==20.11.0
jedi==0.19.1
Jinja2==3.1.4
jmespath==0.10.0
joblib==1.4.0
json5==0.9.24
jsonpointer==2.4
jsonschema==4.23.0
jsonschema-specifications==2023.12.1
jupyter==1.0.0
jupyter-console==6.6.3
jupyter-events==0.10.0
jupyter-lsp==2.2.4
jupyter_client==8.6.1
jupyter_core==5.7.2
jupyter_server==2.13.0
jupyter_server_terminals==0.5.3
jupyterlab==4.1.5
jupyterlab_pygments==0.3.0
jupyterlab_server==2.25.4
jupyterlab_widgets==3.0.10

kiwisolver==1.4.5
langcodes==3.3.0
lazy_loader==0.4
libclang==18.1.1
librosa==0.10.1
linkify-it-py==2.0.3
llvmlite==0.42.0
Markdown==3.6
markdown-it-py==2.2.0
MarkupSafe==2.1.5
marshmallow==3.21.1
matplotlib==3.8.3
matplotlib-inline==0.1.7
mdit-py-plugins==0.3.3
mdurl==0.1.2
mistune==3.0.2

mkl-random
mkl-service==2.4.0
ml-dtypes==0.3.2
mmcv==2.1.0
mmengine==0.10.3
model-index==0.1.11
moviepy==1.0.3
msgpack==1.0.8
multidict==6.0.5
multiprocess==0.70.16
multiprocessing-logging==0.3.4
murmurhash==1.0.10
namex==0.0.7
nbclient==0.10.0
nbconvert==7.16.2
nbformat==5.10.3
nest-asyncio==1.6.0
ninja==1.11.1.1
notebook==7.1.2
notebook_shim==0.2.4
numba==0.59.1

oauthlib==3.2.2
omegaconf==2.3.0
opencv-python==4.7.0.72
#opendatalab==0.0.10
#openmim==0.3.9
#openxlab==0.0.36
opt-einsum==3.3.0
optree==0.10.0
ordered-set==4.1.0
orjson==3.9.15
oss2==2.17.0
overrides==7.7.0
packaging==24.1
pandas==1.5.3
pandocfilters==1.5.1
parso==0.8.4
PasteDeploy==3.1.0
pathtools==0.1.2
pbkdf2==1.3
peft==0.3.0
pexpect==4.9.0
Pillow==9.5.0
plaster==1.1.2
plaster-pastedeploy==1.0.1
platformdirs==4.3.6
pooch==1.8.1
portalocker==2.8.2
preshed==3.0.9
proglog==0.1.10
prometheus_client==0.20.0
prompt_toolkit==3.0.47
protobuf==4.25.3
psutil==6.0.0
ptyprocess==0.7.0
pure_eval==0.2.3
py-cpuinfo==9.0.0
pyarrow==16.0.0
pyarrow-hotfix==0.6
pyasn1==0.6.0
pycocoevalcap==1.2
pycocotools==2.0.7
pycparser==2.22
pycryptodome==3.20.0
pydantic==2.6.4

pydub==0.25.1
Pygments==2.18.0
pynvml==11.5.0
pyparsing==3.1.2
pyramid==2.0.2
pyramid-mailer==0.15.1
pysubs2==1.7.3
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
python-json-logger==2.0.7
python-multipart==0.0.9
python3-openid==3.2.0
pytube==15.0.0
pytz==2023.4
PyYAML==6.0.2
pyzmq==25.1.2
qtconsole==5.5.1
QtPy==2.4.1
redis==5.0.3
referencing==0.35.1
regex==2023.12.25
repoze.sendmail==4.4.1
requests==2.32.3

rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==13.4.2
rpds-py==0.20.0
rsa==4.7.2
s3transfer==0.10.1  # 确保兼容旧版本
safetensors==0.4.2
scikit-learn==1.4.1.post1
scipy==1.10.1
semantic-version==2.10.0
Send2Trash==1.8.2
sentencepiece==0.1.99
sentry-sdk==1.42.0
setproctitle==1.3.3
six==1.16.0
smart-open==6.4.0
smmap==5.0.1
sniffio==1.3.1
soundfile==0.12.1
soupsieve==2.5
soxr==0.3.7
spacy==3.7.4
spacy-legacy==3.0.12
spacy-loggers==1.0.5
SQLAlchemy==2.0.28
srsly==2.4.8
stack-data==0.6.3
starlette==0.36.3
tabulate==0.9.0

tensorboard-data-server==0.7.2

tensorflow-io-gcs-filesystem==0.36.0
termcolor==2.3.0
terminado==0.18.1
thinc==8.2.3
threadpoolctl==3.4.0
timm==0.6.12
tinycss2==1.2.1
tokenizers==0.19.1
tomli==2.0.1
toolz==0.12.1
tornado==6.4.1
tqdm==4.65.2
traitlets==5.14.3
transaction==4.0
transformers==4.40.0
translationstring==1.4
typer==0.9.4
types-python-dateutil==2.9.0.20240316

tzdata==2024.1
uc-micro-py==1.0.3
uri-template==1.3.0

uvicorn==0.28.0
velruse==1.1.1
venusian==3.1.0
wandb==0.14.0
wasabi==1.1.2
wcwidth==0.2.13
weasel==0.3.4

webdataset==0.2.86
webencodings==0.5.1
WebOb==1.8.7
websocket-client==1.7.0
websockets==11.0.3
webvtt-py==0.5.1
Werkzeug==3.0.1
widgetsnbextension==4.0.10
wrapt==1.16.0
WTForms==3.1.2
wtforms-recaptcha==0.3.2
xxhash==3.4.1
yacs==0.1.8
yapf==0.40.2
yarl==1.9.4
zipp==3.20.2
zope.deprecation==5.0
zope.interface==6.2
zope.sqlalchemy==3.1

**Comment by LanXingXuan** [comment]
Thank you for your support of the TimeSuite project and your contributions to the open-source community.

We deeply apologize for the inconvenience caused by the dependency conflicts you encountered while configuring the project environment.

We sincerely appreciate your willingness to share the final process of configuring the environment, helping future users save time. Such a spirit of sharing not only revitalizes the open-source community but also benefits many developers.

If you encounter any other issues or have suggestions while using the TimeSuite project, please feel free to contact us. We are committed to working with you to improve the project.

Thank you once again for your support and contributions!

**Comment by LiangMeng89** [comment]
> Thank you for your support of the TimeSuite project and your contributions to the open-source community.
> 
> We deeply apologize for the inconvenience caused by the dependency conflicts you encountered while configuring the project environment.
> 
> We sincerely appreciate your willingness to share the final process of configuring the environment, helping future users save time. Such a spirit of sharing not only revitalizes the open-source community but also benefits many developers.
> 
> If you encounter any other issues or have suggestions while using the TimeSuite project, please feel free to contact us. We are committed to working with you to improve the project.
> 
> Thank you once again for your support and contributions!

Thank you for your reply, it's a pleasure! I'm testing your demo video and my video data. I will ask you if I have any issues.
